### DISCUSSION TITLE
Hiding routes from OpenAPI Schema Documentation

### DISCUSSION THREAD

**Discussion by DaelonSuzuka** [zauberzeug/nicegui#1188] [Ideas / Feature Requests]
I'm working on an application that has both pages and a REST API, and I want to be able to direct other developers to the generated docs page that FastAPI makes.

Currently this page contains an entry for every `@ui.page()` decorated endpoint, which are not relevant to my target audience.

It also contains endpoints for `/_nicegui/{version}/libraries/{key}` and `/_nicegui/{version}/components/{key}`, which are probably not relevant for *anybody's* target audience.

Individual pages can be hidden via `ui.page(..., include_in_schema=False)`, but this is sort of annoying because I'd have to do it for every page.

I think both of these should be hidden from the API docs by default, and a new option should be added to `ui.run()` letting the user specify the desired level of documentation. Maybe something like:

```py
ui.run(endpoint_documentation=None) # new default
ui.run(endpoint_documentation='all') # everything, current behavior
ui.run(endpoint_documentation='internal pages') # different subcategories can be enabled via a space separated string
```

This is similar to how the (now removed) exclude option worked, it provides granular configurability, and can allow creating new categories in the future(not that I can imagine what those would be).

**Comment by rodja** [comment]
That would indeed be a nice addition. Anybody interested in creating a pull request?

**Reply by DaelonSuzuka** [reply]
I might give it a shot.

**Comment by ProbablyBrian** [comment]
This is very similar to what I was drawing attention to in #1175, though I'd like to be able to hide all of the docs (/docs, /redoc, /openapi.json).  More granular control on a route-by-route basis would be great as well.

**Reply by DaelonSuzuka** [reply]
Interesting, I'll see if I can implement that.

**Reply by DaelonSuzuka** [reply]
I checked it out and the process of mounting an app to another app is complicated enough that I couldn't get it working tonight. I'll revisit this when I get more experience poking around the internals.

**Comment by DaelonSuzuka** [comment]
With the merge of #1196, this feature is now implemented.
